src/main/kotlin/nl/info/client/brp/BrpClientService.kt

```diff
@@ -104,18 +106,24 @@ class BrpClientService @Inject constructor(
         auditEvent: String,
         defaultPurpose: String?,
         extractPurpose: (ZaakafhandelParameters) -> String?
-    ): String? = Regex("""ZAAK-\d{4}-\d+""")
-        .find(auditEvent)
-        ?.value
-        ?.let { zaakIdentificatie ->
-            runCatching {
-                val zaak = zrcClientService.readZaakByID(zaakIdentificatie)
-                val parameters = zaakafhandelParameterService.readZaakafhandelParameters(
-                    zaak.zaaktype.extractUuid()
-                )
-                extractPurpose(parameters)
-            }.getOrNull()
-        } ?: defaultPurpose
+    ): String? =
+        auditEvent
+            .also { LOG.info("Resolving purpose for audit event: $it") }
+            .let { """ZAAK-\d{4}-\d+""".toRegex().find(it)?.value }
+            ?.runCatching {
+                zrcClientService.readZaakByID(this)
+                    .zaaktype.extractUuid()
+                    .let(zaakafhandelParameterService::readZaakafhandelParameters)
+                    .let(extractPurpose)
+            }?.onFailure {
+                LOG.warning("Failed to resolve purpose from audit event '$auditEvent': ${it.message}")
+            }?.getOrElse {
+                LOG.warning("Using default purpose '$defaultPurpose' for audit event '$auditEvent'")
+                null
+            } ?: run {
+            LOG.info("No purpose found in audit event '$auditEvent', using default purpose '$defaultPurpose'")
+            defaultPurpose
+        }
```

Comment:
Not sure how this is formatted correctly. Is there no Detekt error on this bit? It seems to me that these lines should be indented another 4 spaces. Otherwise the `?:` might be better placed on the next line...